indico/modules/users/models/export.py

```diff
@@ -0,0 +1,133 @@
+from indico.core.db.sqlalchemy import PyIntEnum, UTCDateTime
+from indico.util.date_time import now_utc
+from indico.util.enum import RichIntEnum
+from indico.util.i18n import L_
+
+
+class DataExportRequestState(RichIntEnum):
+    __titles__ = [L_('Pending'), L_('Running'), L_('Success'), L_('Failed'), L_('Expired')]
+    pending = 0  # Default value when there is no request
+    running = 1
+    success = 2
+    failed = 3
+    expired = 4  # The associated file has been deleted
+
+
+class DataExportOptions(RichIntEnum):
+    __titles__ = [L_('Personal data'), L_('Settings'), L_('Minutes & Contributions'), L_('Registrations'),
+                  L_('Room booking data'), L_('Abstracts & Papers'), L_('Survey submissions'),
+                  L_('Attachments & Materials'), L_('Editables'), L_('Miscellaneous')]
+    personal_data = 0
+    settings = 1
+    contribs = 2
+    registrations = 3
+    room_booking = 4
+    abstracts_papers = 5
+    survey_submissions = 6
+    attachments = 7
+    editables = 8
+    misc = 9
+
+
+class DataExportRequest(db.Model):
+    __tablename__ = 'data_export_requests'
+    __table_args__ = (db.UniqueConstraint('user_id'),
+                      # If the state is 'success', there must be a file attached
+                      db.CheckConstraint(f'(state != {DataExportRequestState.success}) OR (file_id IS NOT NULL)',
+                                         'success_has_file'),
+                      {'schema': 'users'})
+
+    id = db.Column(
+        db.Integer,
+        primary_key=True
+    )
+    #: The user id of the requestor
+    user_id = db.Column(
+        db.ForeignKey('users.users.id', ondelete='CASCADE'),
+        nullable=False,
+        index=True
+    )
+    #: The id of the generated zip file
+    file_id = db.Column(
+        db.ForeignKey('indico.files.id'),
+        nullable=True,
+        index=True
+    )
+    #: The date and time the request was created
+    requested_dt = db.Column(
+        UTCDateTime,
+        default=now_utc,
+        nullable=False
+    )
+    #: The items which where requested to be exported
+    selected_options = db.Column(
+        ARRAY(sa.Enum(DataExportOptions)),
+        nullable=False,
+        default=[],
+    )
+    # The request state
+    state = db.Column(
+        PyIntEnum(DataExportRequestState),
+        nullable=False,
+        default=DataExportRequestState.pending
+    )
+    # Whether the export archive exceeded config.MAX_DATA_EXPORT_SIZE
+    max_size_exceeded = db.Column(
+        db.Boolean,
+        nullable=False,
+        default=False
+    )
+
+    user = db.relationship(
+        'User',
+        lazy=False,
+        backref=db.backref(
+            'data_export_request',
+            lazy=True,
+            uselist=False,
+            cascade='all, delete-orphan',
+            passive_deletes=True
+        )
+    )
+
+    file = db.relationship(
+        'File',
+        lazy=False,
+        cascade='all, delete-orphan',
```

Comment:
I'm not sure this is a good idea. We usually do not want to simply delete File objects from the DB just because the thing referencing them got deleted - this would leave the file orphaned on the storage backend, with no metadata in indico.

You could copy the relationship config from some other place like this one and change the backref+attribute names:

```python
    custom_boa = db.relationship(
        'File',
        lazy=True,
        backref=db.backref(
            'custom_boa_of',
            lazy=True
        )
    )
```